### PR TITLE
DAC6-1749 | Added UTR page and brought in dependencies for future page creation

### PR DESCRIPTION
**Pull request by matthewharris1**

### CODE REVIEW

File: app/pages/PageLists.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import models.UserAnswers
+
+import scala.util.Try
+
+object PageLists {
+
+  val removePage: (Try[UserAnswers], QuestionPage[_]) => Try[UserAnswers] =
+    (ua: Try[UserAnswers], page: QuestionPage[_]) => ua.flatMap(_.remove(page))
+
+  val businessWithIDPages =
```
Comment (ashwinirudrappa) [inline app/pages/PageLists.scala:28]: @matthewharris1 Lets not go ahead of ourselves, CYA navigation ticket should bring these changes

File: test/pages/DoYouHaveUniqueTaxPayerReferencePageSpec.scala [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+//  "doYouHaveUTRPage" - {
+//
+//    beRetrievable[Boolean](doYouHaveUTRPage)
+//
+//    beSettable[Boolean](doYouHaveUTRPage)
+//
+//    beRemovable[Boolean](doYouHaveUTRPage)
+//  }
+//
+//  "cleanup" - {
```
Comment (ashwinirudrappa) [inline test/pages/DoYouHaveUniqueTaxPayerReferencePageSpec.scala:49]: @matthewharris1 Please remove the cleanup code, let's bring it when needed

File: app/controllers/actions/IdentifierAction.scala [hunk 3]
```diff
@@ -20,55 +20,64 @@ import com.google.inject.Inject
...
-  }
+  override def apply(): ActionBuilder[IdentifierRequest, AnyContent] with ActionFunction[Request, IdentifierRequest] =
+    new AuthenticatedIdentifierActionWithRegime(authConnector, config, parser)
 }
 
-class SessionIdentifierAction @Inject()(
-                                         val parser: BodyParsers.Default
-                                       )
-                                       (implicit val executionContext: ExecutionContext) extends IdentifierAction {
+class AuthenticatedIdentifierActionWithRegime @Inject() (
```
Comment (soljeggers) [inline app/controllers/actions/IdentifierAction.scala:52]: could we have a single AuthenicatedIdentifierAction class?

File: app/controllers/actions/IdentifierAction.scala [hunk 4]
```diff
@@ -20,55 +20,64 @@ import com.google.inject.Inject
...
-      }.getOrElse(throw new UnauthorizedException("Unable to retrieve internal Id"))
-    } recover {
-      case _: NoActiveSession =>
-        Redirect(config.loginUrl, Map("continue" -> Seq(config.loginContinueUrl)))
-      case _: AuthorisationException =>
-        Redirect(routes.UnauthorisedController.onPageLoad)
-    }
-  }
+  override def apply(): ActionBuilder[IdentifierRequest, AnyContent] with ActionFunction[Request, IdentifierRequest] =
+    new AuthenticatedIdentifierActionWithRegime(authConnector, config, parser)
```
Comment (soljeggers) [inline app/controllers/actions/IdentifierAction.scala:49]: Do we need to instantiate a separate class here? we aren't going to be using a regime in this service.

File: app/controllers/actions/DataRequiredAction.scala [hunk 5]
```diff
@@ -16,25 +16,29 @@
...
 import play.api.mvc.Results.Redirect
 import play.api.mvc.{ActionRefiner, Result}
 
+import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class DataRequiredActionImpl @Inject()(implicit val executionContext: ExecutionContext) extends DataRequiredAction {
-
-  override protected def refine[A](request: OptionalDataRequest[A]): Future[Either[Result, DataRequest[A]]] = {
+class DataRequiredActionImplWithRegime(implicit val executionContext: ExecutionContext) extends ActionRefiner[OptionalDataRequest, DataRequest] {
```
Comment (soljeggers) [inline app/controllers/actions/DataRequiredAction.scala:27]: Change name back to previous - no regime being used in this service

**Review by ashwinirudrappa — APPROVED** [review]

**Pull request merged**